operator/cmd/mesh/manifest_shared_test.go

```diff
@@ -28,6 +62,233 @@ const (
+			}
+			objs, err = parseObjectSetFromManifest(m)
+			if err != nil {
+				return nil, err
+			}
+		case cmdApply:
+			objs, err = fakeApplyManifest(inFile, flags, chartSource)
+		case cmdController:
+			objs, err = fakeControllerReconcile(inFile, chartSource)
+		default:
+		}
+		if err != nil {
+			return nil, err
+		}
+		out[cmd] = objs
+	}
+
+	return out, nil
+}
+
+// fakeApplyManifest runs manifest apply. It is assumed that
+func fakeApplyManifest(inFile, flags string, chartSource chartSourceType) (*ObjectSet, error) {
+	inPath := filepath.Join(testDataDir, "input", inFile+".yaml")
+	manifest, err := runManifestCommand("apply", []string{inPath}, flags, chartSource)
+	if err != nil {
+		return nil, fmt.Errorf("error %s: %s", err, manifest)
+	}
+	return NewObjectSet(getAllIstioObjects()), nil
+}
+
+// fakeApplyExtraResources applies any extra resources for the given test name.
+func fakeApplyExtraResources(inFile string) error {
+	reconciler, err := helmreconciler.NewHelmReconciler(testClient, testRestConfig, nil, nil)
+	if err != nil {
+		return err
+	}
+
+	if rs, err := readFile(filepath.Join(testDataDir, "input-extra-resources", inFile+".yaml")); err == nil {
+		if err := applyWithReconciler(reconciler, rs); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func fakeControllerReconcile(inFile string, chartSource chartSourceType) (*ObjectSet, error) {
+	l := clog.NewDefaultLogger()
+	_, iops, err := GenerateConfig(
+		[]string{inFileAbsolutePath(inFile)},
+		[]string{"installPackagePath=" + string(chartSource)},
+		false, testRestConfig, l)
+	if err != nil {
+		return nil, err
+	}
+
+	crName := installedSpecCRPrefix
+	if iops.Revision != "" {
+		crName += "-" + iops.Revision
+	}
+	iop, err := translate.IOPStoIOP(iops, crName, v1alpha1.Namespace(iops))
+	if err != nil {
+		return nil, err
+	}
+	iop.Spec.InstallPackagePath = string(chartSource)
+
+	if err := createNamespace(testK8Interface, iop.Namespace); err != nil {
+		return nil, err
+	}
+
+	reconciler, err := helmreconciler.NewHelmReconciler(testClient, testRestConfig, iop, nil)
+	if err != nil {
+		return nil, err
+	}
+	if err := fakeInstallOperator(reconciler, chartSource, iop); err != nil {
+		return nil, err
+	}
+
+	if _, err := reconciler.Reconcile(); err != nil {
+		return nil, err
+	}
+
+	return NewObjectSet(getAllIstioObjects()), nil
+}
+
+// fakeInstallOperator installs the operator manifest resources into a cluster using the given reconciler.
+// The installation is for testing with a kubebuilder fake cluster only, since no functional Deployment will be
+// created.
+func fakeInstallOperator(reconciler *helmreconciler.HelmReconciler, chartSource chartSourceType, iop proto.Message) error {
+	ocArgs := &operatorCommonArgs{
+		manifestsPath:     string(chartSource),
+		istioNamespace:    istioDefaultNamespace,
+		watchedNamespaces: istioDefaultNamespace,
+		operatorNamespace: operatorDefaultNamespace,
+		hub:               "foo",
+		tag:               "bar",
```

Comment:
the `hub` and `tag` here are just placeholder?

Reply:
yes, i've added a comment.